{
  "packageName": "java.awt.print",
  "simpleName": "PrinterIOException",
  "moduleName": "java.desktop",
  "type": "CLASS",
  "superClass": {
    "name": "java.awt.print.PrinterException",
    "type": "Class"
  },
  "members": [
    {
      "returnType": {
        "name": "java.io.IOException",
        "type": "Class"
      },
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "getIOException",
      "comment": "Returns the IOException that terminated the print job. This method predates the general-purpose exception chaining facility. The Throwable.getCause() method is now the preferred means of obtaining this information.",
      "tagMap": {
        "see": [
          "\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/io/IOException.html\" title\u003d\"class in java.io\"\u003e\u003ccode\u003eIOException\u003c/code\u003e\u003c/a\u003e"
        ],
        "return": [
          "the \n\u003ccode\u003eIOException\u003c/code\u003e that terminated the print job."
        ]
      }
    },
    {
      "returnType": {
        "name": "java.lang.Throwable",
        "type": "Class"
      },
      "parameters": [],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "name": "getCause",
      "comment": "Returns the cause of this exception (the IOException that terminated the print job).",
      "tagMap": {
        "Overrides:": [
          "\u003ccode\u003e\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Throwable.html#getCause()\"\u003egetCause\u003c/a\u003e\u003c/code\u003e\u0026nbsp;in class\u0026nbsp;\n\u003ccode\u003e\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/lang/Throwable.html\" title\u003d\"class in java.lang\"\u003eThrowable\u003c/a\u003e\u003c/code\u003e"
        ],
        "return": [
          "the cause of this exception."
        ],
        "since": [
          "1.4"
        ]
      }
    },
    {
      "parameters": [
        {
          "type": {
            "name": "java.io.IOException",
            "type": "Class"
          },
          "field": {
            "name": "exception"
          }
        }
      ],
      "memberType": "JavaMethod",
      "modifiers": 1,
      "comment": "Constructs a new PrinterIOException with the string representation of the specified IOException.",
      "tagMap": {
        "param": [
          "\u003ccode\u003eexception\u003c/code\u003e - the specified \n\u003ccode\u003eIOException\u003c/code\u003e"
        ]
      }
    }
  ],
  "memberType": "JavaClass",
  "modifiers": 1,
  "name": "java.awt.print.PrinterIOException",
  "comment": "The \n\u003ccode\u003ePrinterIOException\u003c/code\u003e class is a subclass of \n\u003ca href\u003d\"PrinterException.html\" title\u003d\"class in java.awt.print\"\u003e\u003ccode\u003ePrinterException\u003c/code\u003e\u003c/a\u003e and is used to indicate that an IO error of some sort has occurred while printing. \n\u003cp\u003eAs of release 1.4, this exception has been retrofitted to conform to the general purpose exception-chaining mechanism. The \"\u003ccode\u003eIOException\u003c/code\u003e that terminated the print job\" that is provided at construction time and accessed via the \u003ca href\u003d\"#getIOException()\"\u003e\u003ccode\u003egetIOException()\u003c/code\u003e\u003c/a\u003e method is now known as the \u003ci\u003ecause\u003c/i\u003e, and may be accessed via the \u003ca href\u003d\"../../../../java.base/java/lang/Throwable.html#getCause()\"\u003e\u003ccode\u003eThrowable.getCause()\u003c/code\u003e\u003c/a\u003e method, as well as the aforementioned \"legacy method.\"\u003c/p\u003e",
  "tagMap": {
    "see": [
      "\u003ca href\u003d\"https://docs.oracle.com/en/java/javase/16/docs/api/serialized-form.html#java.awt.print.PrinterIOException\"\u003eSerialized Form\u003c/a\u003e"
    ]
  }
}